source/pool/test/unit.js

```diff
@@ -1,607 +1,653 @@
+        pool.connect(alice).withdraw(
+          [
+            {
+              root: getRoot(tree),
+              score: ALICE_SCORE,
+              proof,
+            },
+          ],
+          feeToken.address
+        )
+      ).to.be.revertedWith('CLAIM_ALREADY_MADE')
+    })
+
+    it('withdraw reverts with claim already made on previous pool contract', async () => {
+      await feeToken.mock.balanceOf.returns('100000')
+      await feeToken.mock.transfer.returns(true)
+
+      const root = getRoot(tree)
+
+      await pool
+        .connect(deployer)
+        .setClaimed(root, [alice.address, bob.address])
+
+      let proof = getProof(tree, soliditySha3(bob.address, BOB_SCORE))
+
+      await expect(
+        pool.connect(bob).withdraw(
+          [
+            {
+              root: getRoot(tree),
+              score: BOB_SCORE,
+              proof,
+            },
+          ],
+          feeToken.address
+        )
+      ).to.be.revertedWith('CLAIM_ALREADY_MADE')
+
+      proof = getProof(tree, soliditySha3(alice.address, ALICE_SCORE))
+
+      await expect(
+        pool.connect(alice).withdraw(
+          [
+            {
+              root: getRoot(tree),
+              score: ALICE_SCORE,
+              proof,
+            },
+          ],
+          feeToken.address
+        )
+      ).to.be.revertedWith('CLAIM_ALREADY_MADE')
+    })
+
+    it('withdraw reverts with invalid proof', async () => {
+      await feeToken.mock.transfer.returns(true)
+
+      const root = getRoot(tree)
+      await pool.connect(deployer).enable(root)
+
+      const proof = getProof(tree, soliditySha3(alice.address, ALICE_SCORE))
+
+      await expect(
+        pool.connect(alice).withdraw(
+          [
+            {
+              root: getRoot(tree),
+              score: BOB_SCORE,
+              proof,
+            },
+          ],
+          feeToken.address
+        )
+      ).to.be.revertedWith('PROOF_INVALID')
+    })
+
+    it('withdrawWithRecipient success', async () => {
+      await feeToken.mock.balanceOf.returns('100000')
+      await feeToken.mock.transfer.returns(true)
+
+      const root = getRoot(tree)
+      await pool.connect(deployer).enable(root)
+
+      const proof = getProof(tree, soliditySha3(alice.address, ALICE_SCORE))
+      const withdrawMinimum = 0
+      await expect(
+        await pool.connect(alice).withdrawWithRecipient(
+          [
+            {
+              root: getRoot(tree),
+              score: ALICE_SCORE,
+              proof,
+            },
+          ],
+          feeToken.address,
+          withdrawMinimum,
+          bob.address
+        )
+      ).to.emit(pool, 'Withdraw')
+
+      const isClaimed = await pool.claimed(root, alice.address)
+      expect(isClaimed).to.equal(true)
+    })
+
+    it('withdrawWithRecipient reverts with minimumAmount not met', async () => {
+      await feeToken.mock.balanceOf.returns('100000')
+      await feeToken.mock.transfer.returns(true)

```

Comment:
This was already present from a previous commit, but in review I realize it is not needed, so I suggest removing this line.
`await feeToken.mock.transfer.returns(true)`